### PR TITLE
Login and Signup System

### PR DESCRIPTION
**Pull request by Elben85**

### CODE REVIEW

**Comment by sugiyem** [comment]
1. The styling somehow makes the app to keeps on crashing. 
2. Every username is considered as an invalid username.
3. Valid email is considered as an invalid email.
4. When try to login using unregistered email, it will throw a console error instead of an alert.

**Comment by sugiyem** [comment]
It seems like all the error (except the styling) is caused by using onChange instead of onChangeText in TextInput component.

File: src/screens/LoginScreen.js [hunk 1]
```diff
@@ -1,14 +1,97 @@
...
+  }
+
+  return (
+    <SafeAreaView style={styles.Container}>
+      <Text> Log In </Text>
+      <Text> email </Text>
+      <TextInput
+        placeholder="email"
+        value={credentials.email}
+        onChange={(e) => handleChange(e, "email")}
```
Comment (sugiyem) [inline src/screens/LoginScreen.js:59]: Should be changed to onChangeText={(text) => handleChange(text, "email")} or something like that

File: src/screens/LoginScreen.js [hunk 2]
```diff
@@ -1,14 +1,97 @@
...
+  password: ""
+};
+
+const LoginScreen = ({ navigation }) => {
+  const [credentials, setCredentials] = useState(initialState);
+
+  function handleChange(event, name) {
+    setCredentials({
+      ...credentials,
+      [name]: event.target.value
```
Comment (sugiyem) [inline src/screens/LoginScreen.js:23]: Should be changed to "[name]: text" where text is the first parameter of the function.

File: src/screens/LoginScreen.js [hunk 3]
```diff
@@ -1,14 +1,97 @@
...
+  }
+
+  console.log(credentials);
+
+  async function handleSubmit() {
+    await firebase
+      .auth()
+      .signInWithEmailAndPassword(credentials.email, credentials.password)
+      .catch((e) => {
+        console.error(e);
```
Comment (sugiyem) [inline src/screens/LoginScreen.js:34]: Should give an alert instead

File: src/screens/LoginScreen.js [hunk 4]
```diff
@@ -1,14 +1,97 @@
...
+    console.log(currentUser);
+
+    if (currentUser === null) {
+      Alert.alert("Incorrect Email / Password");
+      await firebase.auth().signOut();
+    } else if (!currentUser.emailVerified) {
+      Alert.alert("Please Verify Your Email Before Logging In");
+      await firebase.auth().signOut();
+    } else {
+      navigation.navigate("Dashboard");
```
Comment (sugiyem) [inline src/screens/LoginScreen.js:48]: If we do this, the user can go back to login page after successful sign in. Should be changed to `navigation.replace("Dashboard")` instead.

Comment (Elben85) [reply src/screens/LoginScreen.js:48]: Note: navigation.navigate("Signup") and Login  should also be change for the same reason

**Comment by Elben85** [comment]
> 1. The styling somehow makes the app to keeps on crashing.

Styling error cause by heigh styling

**Comment by sugiyem** [comment]
Looks good to me

**Pull request merged**